core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -165,6 +176,63 @@ protected void beforeDisconnect() {
     // Empty implementation.
   }
 
+  /**
+   * Handshakes with meta master.
+   */
+  private synchronized void handshake() throws AlluxioStatusException {
+    if (isConnected() || sHandshakeComplete.get()) {
+      return;
+    }
+    mAddress = getAddress();
+    LOG.info("Alluxio client (version {}) is trying to bootstrap-connect with {}",
+        RuntimeConstants.VERSION, mAddress);
+    // A plain socket transport to bootstrap
+    TTransport baseTransport = ThriftUtils.createThriftSocket(mAddress);
+    TTransport transport = new BootstrapClientTransport(baseTransport);
+    TProtocol protocol = ThriftUtils.createThriftProtocol(transport,
+        Constants.META_MASTER_SERVICE_NAME);
+    List<ConfigProperty> clusterConfig;
+    try {
+      transport.open();
+      MetaMasterClientService.Client client = new MetaMasterClientService.Client(protocol);
+      clusterConfig = client.getConfiguration(new GetConfigurationTOptions())
+          .getConfigList()
+          .stream()
+          .map(ConfigProperty::fromThrift)
```

Comment:
should we filter to only include client configuration?

Reply:
It is actually hard to filter currently.  I had a discussion with @gpang and we think this is perhaps easier to explain to users what's happening. We could have a discussion offline later--I am open to alternative ways.